systests/jaxrs/src/test/java/org/apache/cxf/systest/jaxrs/BookApplication.java

```diff
@@ -106,6 +106,12 @@ public void setDefaultId(List<String> ids) {
         public void aroundWriteTo(WriterInterceptorContext context) throws IOException,
             WebApplicationException {
             context.getHeaders().putSingle("BookWriter", "TheBook");
+            
+            final Object property = context.getProperty("x-book");
+            if (property != null) {
+                context.getHeaders().putSingle("X-Book-Header", property);
```

Comment:
```suggestion
                context.getHeaders().putSingle("X-Book-Response-Header", property);
```

Just a suggestion, but IIUC, the intent is to show that property set from the request filter is propagated to the writer interceptor (in the response flow).  By using a different header in the response, it makes it clear that the test is not just propagating headers from the request back to the response.

Reply:
Completely agree, rewrote the test to reflect its purpose, thanks a lot!